helmfile/charts/prometheus-alerts/templates/alerts/daily-checks.yaml

```diff
@@ -17,15 +18,29 @@ spec:
   groups:
   - name: daily-checks
     rules:
+{{- if .Values.s3BucketPercentLimitAlarm }}
     - alert: S3BucketContentOver{{.Values.s3BucketPercentLimit}}Percent
       annotations:
-        description: Over {{.Values.s3BucketPercentLimit}} percent of the total s3bucket size limit is used.
-        summary: Over {{.Values.s3BucketPercentLimit}} percent of the total s3bucket size limit is used.
+        message: Over {{.Values.s3BucketPercentLimit}} percent of the total s3bucket size limit is used in bucket {{`{{ $labels.bucket }}`}}.
       expr: |-
         (
-          sum(max_over_time(s3_objects_size_sum_bytes[1h])) >= ({{ .Values.s3BucketSizeQuotaGB }} * (1024^3)) * ({{ .Values.s3BucketPercentLimit }} / 100)
+          sum(max_over_time(s3_objects_size_sum_bytes[1h])) by (bucket) >= ({{ .Values.s3BucketSizeQuotaGB }} * (1024^3)) * ({{ .Values.s3BucketPercentLimit }} / 100)
         )
       for: 1h
       labels:
         severity: warning
 {{- end }}
+{{- if .Values.s3BucketNrOfObjectsAlarm }}
+    - alert: S3BucketObjectsOver{{.Values.s3BucketPercentLimit}}Percent
+      annotations:
+        message: Over {{.Values.s3BucketPercentLimit}} percent of the total s3bucket object limit is used in bucket {{`{{ $labels.bucket }}`}}.
+      expr: |-
+        (
+          sum(max_over_time(s3_objects[2h]{bucket!~"{{.Values.s3BucketsToExcludeRegex}}"})) by (bucket) > {{ .Values.s3BucketNrOfObjects }} * ({{ .Values.s3BucketPercentLimit }} / 100)
```

Comment:
Take a quick look at https://helm.sh/docs/chart_template_guide/control_structures/#looping-with-the-range-action
And then I think you should be able to change the variable into a list instead of a string

Reply:
Now this have been added to multiple functions in where i can see it fit! :+1: